File: IOSSearch/IOSSearch/ViewController.m
//
//  ViewController.m
//  IOSSearch
//
//  Created by Jimmy Huynh on 6/2/20.
//  Copyright © 2020 Jimmy Huynh. All rights reserved.
//

#import "ViewController.h"
#import <NCMB/NCMB.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}
- (IBAction)SearchIN:(id)sender {
    NCMBQuery *query = [NCMBQuery queryWithClassName:@"SearchClass"];
    [query whereKey:@"name" containedIn:@[@"name5", @"name2"]];
    [query findObjectsInBackgroundWithBlock:^(NSArray *objects, NSError *error) {
        if (error) {
            NSLog(@"Error %@", error);
        } else {
            NSLog(@"Data %@", objects);
        }
    }];
}
- (IBAction)SearchInArray:(id)sender {
    NCMBQuery *query = [NCMBQuery queryWithClassName:@"SearchClass"];
    [query whereKey:@"name" containedInArrayTo:@[@"name5", @"name2"]];
    [query findObjectsInBackgroundWithBlock:^(NSArray *objects, NSError *error) {
        if (error) {
            NSLog(@"Error %@", error);
        } else {
            NSLog(@"Data %@", objects);
        }
    }];
}


@end

File: SwiftSearch/SwiftSearch/ViewController.swift
//
//  ViewController.swift
//  SwiftSearch
//
//  Created by Jimmy Huynh on 6/2/20.
//  Copyright © 2020 Jimmy Huynh. All rights reserved.
//

import UIKit
import NCMB

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }


    @IBAction func SearchIn(_ sender: Any) {
        var query : NCMBQuery<NCMBObject> = NCMBQuery.getQuery(className: "SearchClass")
        query.where(field: "name", containedIn: ["name5","name2"])
        query.findInBackground(callback: { result in
            switch result {
                case let .success(array):
                    print("result: \(array)")
                case let .failure(error):
                    print("error: \(error)")
            }
        })
    }
    
    @IBAction func SearchInArray(_ sender: Any) {
        var query : NCMBQuery<NCMBObject> = NCMBQuery.getQuery(className: "SearchClass")
        query.where(field: "name", containedInArrayTo: ["name5","name2"])
        query.findInBackground(callback: { result in
            switch result {
                case let .success(array):
                    print("result: \(array)")
                case let .failure(error):
                    print("error: \(error)")
            }
        })
    }
}

